oslabs-beta/apollo-epoch
# Network responses

**Problem**
Network requests were not coming back with the proper responses / cache snapshots due to their async nature

**Solution**
Implemented a network listener and async store to detect and keep network queries out of the timeline until their responses came in

**To Test**
Open the test app and click around. You should see query responses returning in the order we receive their responses. Instantaneously for cache pulls and at the time of return for network requests. Network requests should have attached responses and appropriate cache diffs.
